### PR TITLE
Creare Azure SQL datastore in AML

### PR DESCRIPTION
**Pull request by achalachal2003**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by achalachal2003** [comment]
@bmoore-msft for review thanks!

File: 101-datastore-create-sql/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		},
+		"passwordOrClientSecret": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The password of the database user or service principal's secret."
+			}
+		},
+		"tenantId": {
+			"type": "string",
+			"defaultValue": "[if(equals(parameters('authenticationType'), 'Service principal'), subscription().tenantId, '')]",
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:48]: if this needs to be conditional, move it to the property assignment.

File: 101-datastore-create-sql/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		},
+		"databaseName": {
+			"type": "string",
+			"metadata": {
+				"description": "The database name."
+			}
+		},
+		"authorityUrl": {
+			"type": "string",
+			"defaultValue": "https://login.microsoftonline.com",
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:61]: Are there other authn methods we support?

If we do need this, use the environment() function to get the value (you cannot hardcode it)

File: 101-datastore-create-sql/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			"metadata": {
+				"description": "Optional : Authority url used to authenticate the user. Defaults to https://login.microsoftonline.com"
+			}
+		},
+		"resourceUrl": {
+			"type": "string",
+			"defaultValue": "https://database.windows.net/",
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:68]: use the environment() function to get the value

File: 101-datastore-create-sql/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+			"metadata": {
+				"description": "Optional : Determines what operations will be performed on the database. Defaults to https://database.windows.net/"
+			}
+		},
+		"endpoint": {
+			"type": "string",
+			"defaultValue": "database.windows.net",
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:75]: use the environment function

File: 101-datastore-create-sql/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+			"metadata": {
+				"description": "Optional : The endpoint of the sql server. Defaults to database.windows.net."
+			}
+		},
+		"skipValidation": {
+			"type": "bool",
+			"defaultValue": false,
+			"allowedValues": [
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:83]: remove allowedValues (the type is sufficient)

File: 101-datastore-create-sql/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+			"type": "Microsoft.MachineLearningServices/workspaces/datastores",
+			"name": "[concat(parameters('workspaceName'), '/', parameters('datastoreName'))]",
+			"apiVersion": "2020-05-01-preview",
+			"location": "[parameters('location')]",
+			"properties": {
+				"DataStoreType": "sqldb",
+				"SkipValidation": "[parameters('skipValidation')]",
+				"DatabaseName": "[parameters('databaseName')]",
+				"ServerName": "[parameters('sqlServerName')]",
+				"UserName": "[if(equals(parameters('authenticationType'), 'SQL Authentication'), parameters('userNameOrClientId'), '')]",
```
Comment (bmoore-msft) [inline 101-datastore-create-sql/azuredeploy.json:110]: replace empty strings with json('null')

File: 101-datastore-create-sql/azuredeploy.json (hunk 2 above)
Comment (achalachal2003) [reply 101-datastore-create-sql/azuredeploy.json:61]: We can remove these

**Pull request merged**